### PR TITLE
updated vector_norm function to conform to Array API Standard

### PR DESCRIPTION
**Pull request by kene111**

### CODE REVIEW

**Comment by djl11** [comment]
Thanks a lot for the PR!

I'll need to rebase this to the latest master branch before continuing. Could you please give me [write access](https://stackoverflow.com/questions/35278957/can-i-modify-someone-elses-pull-request-and-push-it-back-in-its-branch/49854013#:~:text=When%20creating%20a%20pull%20request,commit%20changes%20to%20your%20PR.&text=So%20if%20you%20want%20to,to%20give%20you%20this%20permission.) to the PR? 🙂

**Comment by kene111** [comment]
Hello Daniel, "Allow edits by maintainers" has been checked. Do you now have write access to the PR?

**Comment by djl11** [comment]
> Hello Daniel, "Allow edits by maintainers" has been checked. Do you now have write access to the PR?

Yes, thank you!

**Comment by djl11** [comment]
Thanks a lot for your contribution. Could you update the methods to follow the formatting explained in the [contributor guide](https://lets-unify.ai/ivy/contributing.html) in the [docs](https://lets-unify.ai/ivy/)? I also go through some examples in the Array API [tutorial series](https://www.youtube.com/watch?v=HNV4eI0zp4M&list=PLwNuX3xB_tv96FaEUmTirlQLIsnUgvtTl) on YouTube.

Essentially:

- All methods should be defined with `def func_name` explicitly, no lambdas or direct binding.
- All Ivy methods should have type-hints and doc strings
- All backend methods should have type-hints, but no doc strings.
- All Array API methods should be moved to the correct sub-modules and files.

Please check the contributor guide in the docs for more details. Thanks!

Additionally, it may be necessary to resolve new conflicts with master since the PR was made. As a guide for resolving conflicts with upstream, I have created this short [YouTube series](https://www.youtube.com/watch?v=KCMceLBBlgs&list=PLwNuX3xB_tv8Gf2sELkXSqCCWcUq-AbrG).
Once there are no conflicts with upstream and the formatting is all correct, I'll then take a look at the tests and see if it's ready to be merged :)

File: ivy/functional/backends/mxnet/array_api/linear_algebra_extension.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+ # global
+import mxnet as mx
+from typing  import Union, Optional, Tuple, Literal
+
+# local
+inf = float("inf")
+MxArray = mx.ndarray.ndarray.NDArray
+ 
+def vector_norm(x: MxArray,
+                p:Union[int, float, Literal[inf, - inf]] = 2,
```
[the remark this answers is not in the document]
Comment (djl11) [reply ivy/functional/backends/mxnet/array_api/linear_algebra_extension.py:10]: Just add `mx.ndarray.ndarray.NDArray` to the type hint, it's not too long and will be a bit more descriptive in the docs.

File: ivy/functional/backends/mxnet/array_api/linear_algebra_extension.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+ # global
+import mxnet as mx
+from typing  import Union, Optional, Tuple, Literal
+
+# local
+inf = float("inf")
+MxArray = mx.ndarray.ndarray.NDArray
+ 
+def vector_norm(x: MxArray,
```
Comment (djl11) [inline ivy/functional/backends/mxnet/array_api/linear_algebra_extension.py:9]: Just add mx.ndarray.ndarray.NDArray in the type hint directly

**Comment by kene111** [comment]
Hello David, thanks for the corrections and commits you made, would makes some updates  and make a new PR

**Comment by djl11** [comment]
Thanks!

**Pull request merged**